src/components/YearlyDataTab.tsx

```diff
@@ -0,0 +1,70 @@
+import * as React from "react";
+import CopyButton from "./CopyButton";
+import { Button } from "library-simplified-reusable-components";
+import { LibraryData } from "../interfaces";
+
+export interface YearlyDataTabProps {
+  data: {[key: string]: LibraryData[]};
+}
+
+export interface YearlyDataTabState {
+  styled: boolean;
+}
+
+export default class YearlyDataTab extends React.Component<YearlyDataTabProps, YearlyDataTabState> {
+  private dataRef = React.createRef<HTMLUListElement>();
+  constructor(props: YearlyDataTabProps) {
+    super(props);
+    this.state = { styled: true };
+    this.toggleFormatting = this.toggleFormatting.bind(this);
+  }
+  sortByYear(data) {
+    let sortedByYear = {};
+    Object.keys(data).map((name: string) => {
+      let category = data[name];
+      category.forEach((library) => {
+        let year = library.basic_info.timestamp && library.basic_info.timestamp.match(/20\d+/)[0];
+        if (year) {
+          if (sortedByYear[year]) {
+            sortedByYear[year][name].push(library.basic_info.name);
+          } else {
+            sortedByYear[year] = {"production": [], "testing": [], "cancelled": []};
+            sortedByYear[year][name].push(library.basic_info.name);
+          }
+        }
+      });
+    });
+    return(sortedByYear);
+  }
+  toggleFormatting() {
+    this.setState({ styled: !this.state.styled });
+  }
+  render(): JSX.Element {
+    let categories = (year) => <ul>{Object.keys(year).map(c => <li className={this.state.styled ? "stats-category" : ""}><p className="stats-category-name">{c.substr(0, 1).toUpperCase() + c.substr(1)}</p>{names(year[c])}</li>)}</ul>;
```

Comment:
It's hard to read and easily tell what `c` is, but this can be refactored later.

Reply:
It's "category".  Will change.